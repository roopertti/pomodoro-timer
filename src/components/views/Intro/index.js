import React from 'react'
import { motion } from 'framer-motion'

import Blockquote from 'Components/common/Blockquote'
import Button from 'Components/common/Button'
import useStorage from 'Hooks/useStorage'

const variants = {
  hidden: {
    opacity: 0,
    transform: 'translateY(-30px)'
  },
  visible: {
    opacity: 1,
    transform: 'translateY(0px)'
  }
}

/**
 * Returns Intro motion props
 * @param {Number} delay Delay before component turns visible
 */
function getMotionProps (delay) {
  return {
    animate: 'visible',
    initial: 'hidden',
    variants,
    transition: { duration: 1.5, ease: 'easeOut', delay }
  }
}

function Intro () {
  const { completeIntro } = useStorage()

  function handleClick () {
    completeIntro()
  }

  return (
    <section data-testid="intro-view">
      <motion.div {...getMotionProps(1.5)}>
        <Blockquote
          quote="The Pomodoro Technique is a time management method developed by Francesco Cirillo in the late 1980s. The technique uses a timer to break down work into intervals, traditionally 25 minutes in length, separated by short breaks."
          citeUrl="https://en.wikipedia.org/wiki/Pomodoro_Technique"
          caption="Wikipedia"
        />
      </motion.div>
      <motion.div
        className="flex justify-center py-8"
        {...getMotionProps(4)}
      >
        <Button testId="get-started-btn" onClick={handleClick}>Get started!</Button>
      </motion.div>
    </section>
  )
}

export default Intro
