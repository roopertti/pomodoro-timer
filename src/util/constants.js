/* Local Storage keys */
export const STORE_KEY = 'POMODORO_TIMER_DATA_STORAGE'

/* Action types */
export const SET_INTRO_COMPLETE = 'SET_INTRO_COMPLETE'
export const ADD_ENTRY = 'ADD_ENTRY'

/* Timer states */
export const TIMER_STATE = {
  INITIAL: 'INITIAL',
  RUNNING: 'RUNNING',
  STOPPED: 'STOPPED',
  COMPLETE: 'COMPLETE'
}

export const TIMER_DURATIONS = {
  WORK: 1500,
  BREAK: 300,
  LONG_BREAK: 5
}
