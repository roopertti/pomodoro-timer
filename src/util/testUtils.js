import { render } from '@testing-library/react'
import AppWrapper from '../AppWrapper'

function renderWithWrappers (ui, options) {
  return render(
    ui,
    {
      wrapper: AppWrapper,
      ...options
    }
  )
}

export * from '@testing-library/react'

export { renderWithWrappers as render }
