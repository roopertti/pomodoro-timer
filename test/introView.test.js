import React from 'react'
import renderer from 'react-test-renderer'

import { render, waitFor, fireEvent } from 'Util/testUtils'
import Intro from 'Components/views/Intro'
import { STORE_KEY } from 'Util/constants'

describe('Intro view integration tests', () => {
  it('renders and compares snapshot', () => {
    const tree = renderer.create(<Intro />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders Intro view, completes intro', async () => {
    const { getByTestId } = render(<Intro />)
    expect(getByTestId('intro-view')).toBeInTheDocument()
    fireEvent.click(getByTestId('get-started-btn'))
    await waitFor(() => {
      expect(JSON.parse(global.localStorage.getItem(STORE_KEY))).toStrictEqual({ introComplete: true })
    })
  })
})
