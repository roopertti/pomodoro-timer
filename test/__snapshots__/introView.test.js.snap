// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Intro view integration tests renders and compares snapshot 1`] = `
<section
  data-testid="intro-view"
>
  <div
    style={
      Object {
        "opacity": 0,
        "transform": "translateY(-30px)",
      }
    }
  >
    <figure
      className="blockquote"
    >
      <blockquote
        cite="https://en.wikipedia.org/wiki/Pomodoro_Technique"
      >
        <p>
          "The Pomodoro Technique is a time management method developed by Francesco Cirillo in the late 1980s. The technique uses a timer to break down work into intervals, traditionally 25 minutes in length, separated by short breaks."
        </p>
      </blockquote>
      <figcaption>
        -Wikipedia
      </figcaption>
    </figure>
  </div>
  <div
    className="flex justify-center py-8"
    style={
      Object {
        "opacity": 0,
        "transform": "translateY(-30px)",
      }
    }
  >
    <button
      className="btn"
      data-testid="get-started-btn"
      onClick={[Function]}
      style={
        Object {
          "transform": "translateY(0px)",
        }
      }
    >
      Get started!
    </button>
  </div>
</section>
`;
